### DISCUSSION TITLE
Check if FASTLANE_SESSION is still valid

### DISCUSSION THREAD

**Discussion by Tokta** [fastlane/fastlane#17999] [Q&A]
Hello and thank you for all your hard work!
I have a little annoying issue with the expiration of FASTLANE_SESSION. I know there isn't to much to do to generate it automatically with 2FA, but I'm wondering if there is the possibility to check the validity of the current one before trying to submit a new build to AppStoreConnect when the the token is actually needed. Would be nice to use it for a daily job on the CI to receive a notification when it expire.

Anybody has done this before? There is something I've skipped from the spaceship/spaceauth docs that can come in handy?
Have a great day 😃 !

**Comment by dinsen** [comment]
Hi @Tokta 
You could just fire off one of the actions that require a valid session to check it?

**Reply by Tokta** [reply]
Hi Dinsen, sorry for the late response and thank you for your input. I've tried with `download_dsyms` but apparently is not using `FASTLANE_SESSION` because that job succeeded but the deploy after it failed for the expired session. Do you know which action one uses it?

**Comment by Thangaraj-M** [comment]
The session is generated with the spaceauth command. We can run once again the spaceauth command and get the exit status ($?) of command execution. If the status is success then the session is valid. If the status is failed, then the session is expired.

**Reply by shanu2698** [reply]
When using  `fastlane spaceauth --check_session` command even with freshly created and saved session it says:
`No valid session found (<FASTLANE_USER>). Exiting.`

so I would prefer the above solution.

**Comment by anton-plebanovich** [comment]
There is `fastlane spaceauth --check_session` command for that
